### ISSUE TITLE
Look in core for menu management page & apply to our new widget page

### ISSUE THREAD

**Issue by jtsternberg** [WebDevStudios/WordPress-Widgets-Refresh#3]
Initial idea is for widget management page to mimic menu management page.

**Comment by richardtape** [comment]
For reference:

Current widgets screen is output from wp-admin/widgets.php
Current menu admin screen is output from wp-admin/nav-menus.php - the one on which we're thinking about basing the new widgets screen.

The existing widgets.php file is.. a disgrace. It's so old. Anyone who conforms to the DRY principle would, err, "wet" themselves. The new nav-menus.php file is much cleaner, much more documented. It's still not wonderful, but it's a good starting point.

In the existing widgets.php file there are precisely no apply_filters or apply_filters_ref_array calls and a grand total of 3 do_action calls and zero do_action_ref_array calls.

This doesn't bode well for us building this as a plugin from scratch. We'll need some temporary hooks in place basically immediately otherwise we'll be hacking stuff with css and javascript which, in my opinion, if we're going to do this properly, isn't the way forward. It might give us a proof of concept but the widgets.php file needs a complete overhaul, this is very apparent.

I, for one, didn't realise that if your theme doesn't support widgets (I know this is practically impossible now-a-days but still...) you actually get a wp_die() screen. See line 72. We should probably change that along the way, too ;)

We could simply ask for a temporary hook at the top of widgets.php which would allow us to hook in, and then output _completely_ new stuff and then exit at the end - i.e. build from scratch. There's probably a more elegant solution but on first-read-through it seems like a half-viable option at the moment.

**Comment by jtsternberg** [comment]
I would love for now to build a completely separate admin page alongside the real widgets page. I think we could start with ideal markup that way and basically fake it until we come up with a strong plan. When the time comes we can just remove the real widgets page (all via our plugin). I think this gives us the most flexibility. I love where you're going with your thought process. I agree, current page is abysmal. 
—
Sent on mobile. Please excuse typos.

**Comment by richardtape** [comment]
Yeah I think you're right, Justin, adding a completely separate menu page is the way ahead for the time being. Good plan!
